### PR TITLE
quality: First batch of Local and Remote SDK testing

### PR DESCRIPTION
**Pull request by typotter**
🎟️ Fixes FF-3097 towards FF-3085

👯‍♂️ **Related PRs**
- [php-sdk#38](https://github.com/Eppo-exp/php-sdk/pull/38)
- [dot-net-server-sdk#44](https://github.com/Eppo-exp/dot-net-server-sdk/pull/44)
- [node-server-sdk#69](https://github.com/Eppo-exp/node-server-sdk/pull/69)

### Motivation
Changes are often made to the [sdk-test-data](https://github.com/Eppo-exp/sdk-test-data) repository to capture new behaviours, bugs and edge cases. When these changes are pushed to `main`, the SDKs are cloned locally (locally to the github action running) and their respective tests are run. These tests are set up and run by copies of the SDK test workflows - see [sdk-test-data workflow](https://github.com/Eppo-exp/sdk-test-data/blob/main/.github/workflows/test-sdks.yml). There are a number of limitations to this setup:

- Test steps are copied from the SDK’s respective workflows; changes to the SDK test workflows need to be replicated in sdk-test-data and this is not obvious to devs
- The SDK tests are not able to run against in-flight changes to `sdk-test-data`
- When new test-data is committed that breaks an SDK, that breakage is not surfaced in the SDK’s repository until some action triggers its testing workflow (on demand, on pull-request, etc.).

### Description of Changes
_This change_
- First batch of SDKs for POC: **PHP, dotnet and node-server-sdk**
- Local Testing workflow: call the converted SDKs' reusable workflow specifying the local branch used (i.e. branch of a PR)
- Remote Testing workflow: trigger the converted SDKs workflows using the GH API

_External to this Change_
🚀 - Each SDK's testing workflow is enhanced into a [reusable workflow](https://docs.github.com/en/actions/sharing-automations/reusing-workflows), exposing parameters for SDK branch and the sdk-test-data branch to use in testing.

### CODE REVIEW

File: .github/workflows/test-sdks-remote.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      run: |
+        echo "Testing eppo-exp/php-sdk"
+    - name: Run remote workflow
+      uses: convictional/trigger-workflow-and-wait@v1.6.1
+      with:
+        owner: Eppo-exp
+        repo: php-sdk
+        workflow_file_name: run-tests.yml
+        ref: tp/workflows/remote
+        github_token: ${{ secrets.AUTH_TOKEN }}
```
Comment (typotter) [inline .github/workflows/test-sdks-remote.yml:25]: This is currently my PAT and will need to change .

File: .github/workflows/test-sdks-remote.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+name: Test SDKs Remotely
```
Comment (typotter) [inline .github/workflows/test-sdks-remote.yml:1]: This workflow runs on pushes to the main branch of sdk-test-data and triggers the testing workflows to run **in** each SDK repository, thus turning them red/green depending on whether they pass any new test cases added to this repo.

File: .github/workflows/test-sdks.yml [hunk 3]
```diff
@@ -81,31 +81,10 @@ jobs:
...
-        working-directory: ./
-      - name: 'Set up GCP SDK for downloading test data'
-        uses: 'google-github-actions/setup-gcloud@v0'
-      - name: Install sdk dependencies
-        run: yarn --frozen-lockfile
-        working-directory: ./
-      - name: Run tests
-        run: yarn test:unit
-        working-directory: ./
+    uses: Eppo-exp/node-server-sdk/.github/workflows/lint-test-sdk.yml@tp/workflows/remote
```
Comment (typotter) [inline .github/workflows/test-sdks.yml:84]: temporary branch spec _@tp/workflows/remote_ to be removed when SDK PRs are submitted.

File: .github/workflows/test-sdks-remote.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      shell: bash
+      run: |
+        echo "Testing eppo-exp/php-sdk"
+    - name: Run remote workflow
+      uses: convictional/trigger-workflow-and-wait@v1.6.1
+      with:
+        owner: Eppo-exp
+        repo: php-sdk
+        workflow_file_name: run-tests.yml
+        ref: tp/workflows/remote
```
Comment (typotter) [inline .github/workflows/test-sdks-remote.yml:24]: temp branch pointer until respective PR is merged

File: .github/workflows/test-sdks-remote.yml (hunk 1 above)
Comment (leoromanovsky) [reply .github/workflows/test-sdks-remote.yml:25]: Cool will do after merging to main

**Review by leoromanovsky — APPROVED** [review]

File: .github/workflows/test-sdks-remote.yml (hunk 2 above)
Comment (aarsilv) [reply .github/workflows/test-sdks-remote.yml:1]: Do the results factor into the success of _this_ job? Is that what `propagate_failure` does?

File: .github/workflows/test-sdks-remote.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
+name: Test SDKs Remotely
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
```
Comment (aarsilv) [inline .github/workflows/test-sdks-remote.yml:7]: 👍

**Review by aarsilv — APPROVED** [review]
📈

File: .github/workflows/test-sdks-remote.yml (hunk 2 above)
Comment (typotter) [reply .github/workflows/test-sdks-remote.yml:1]: correct.
